### ISSUE TITLE
Fix binsize CI

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#2953]
The binsize CI is consistently getting stuck:

```
...
Writing wasmpkg/wat/yoke_derive.wat 
Writing wasmpkg/wat/zf_derive.wat 
Writing wasmpkg/wat/zv_serde.wat 
[cargo-make] INFO - Running Task: wasm-dcmp
Writing wasmpkg/wasm-decompile/and_list.dcmp 
Writing wasmpkg/wasm-decompile/borrowed_pattern.dcmp 
Writing wasmpkg/wasm-decompile/code_line_diff.dcmp 
Writing wasmpkg/wasm-decompile/derives.dcmp 
Writing wasmpkg/wasm-decompile/elevator_floors.dcmp 
Writing wasmpkg/wasm-decompile/filter_langids.dcmp 
Writing wasmpkg/wasm-decompile/iso_date_manipulations.dcmp 
Writing wasmpkg/wasm-decompile/iso_datetime_manipulations.dcmp 
Writing wasmpkg/wasm-decompile/language_names_hash_map.dcmp 
Writing wasmpkg/wasm-decompile/language_names_lite_map.dcmp 
Writing wasmpkg/wasm-decompile/litemap_bincode.dcmp 
Writing wasmpkg/wasm-decompile/litemap_postcard.dcmp 
Writing wasmpkg/wasm-decompile/make.dcmp 
Writing wasmpkg/wasm-decompile/make_var.dcmp 
Error: The operation was canceled.
```

**Comment by robertbastian** [comment]
Tried reenabling it, but the issue is: `icu_provider/datagen` depends on `dhat` which doesn't compile on wasm (some dependency's build script panics). However as we build the examples with workspace-level feature resolution, it gets built for some reason.

Solution A: move `HeapStatsMarker` out of `icu_provider`. `dhat` can be a dev-dependency of `icu_datagen`, not a (feature-gated) runtime dependency of `icu_provider`.

Solution B: don't use workspace-level feature resolution, instead build each example individually.

**Comment by sffc** [comment]
It is intended that `dhat` is not always enabled and can be disabled, here:

https://github.com/unicode-org/icu4x/blob/main/tools/benchmark/macros/Cargo.toml

**Comment by robertbastian** [comment]
The binsize bench has nothing to do with that macro, it just builds the examples as wasm. dhat gets pulled in through icu_provider

**Comment by sffc** [comment]
Yeah, okay, hmm. Trying to figure out why this wasn't a problem before. This may be a consequence of making more features on `icu_datagen` be default features.

**Comment by robertbastian** [comment]
`dhat` is a separate issue, after #3138 it still freezes on `wasm-decompile wasmpkg/make_var.wasm`

**Comment by sffc** [comment]
I don't mind if we stop running wasm-decompile in CI because it does not tell us anything about binsize
